File: Java/프로그래머스/1/42748. K번째수/K번째수.java
import java.util.Arrays;

class Solution {
    static private int getResult(int[] array, int[] command) {
        int i = command[0]-1;
        int j = command[1]-1;
        int k = command[2]-1;
        
        int[] result = new int[j-i+1];
        int resultIndex = 0;
        for (int index = i; index<=j; index++) {
            result[resultIndex++] = array[index];
        }
        
        Arrays.sort(result);
        
        return result[k];        
    }
    
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int resultIndex = 0;
        
        for (int[] command : commands) {
            answer[resultIndex++] = getResult(array, command);
        }
        return answer;
    }
}

File: Swift/프로그래머스/1/42748. K번째수/K번째수.swift
import Foundation

func getResult(_ array: [Int], _ command: [Int]) -> Int {
    let i = command[0]-1
    let j = command[1]-1
    let k = command[2]-1
    
    var result: [Int] = .init(repeating: 0, count: j-i+1)
    var resultIndex = 0
    for index in i...j {
        result[resultIndex] = array[index]
        resultIndex += 1
    }
    
    result.sort()
    
    return result[k]
}
func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    var answer: [Int] = .init(repeating: 0, count: commands.count)
    var resultIndex = 0
    for command in commands {
        answer[resultIndex] = getResult(array, command)
        resultIndex += 1
    }
    return answer
}